extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -42,14 +43,23 @@
 /**
  * {@link SeekableChannelsProvider} implementation that is used to fetch objects from an S3-compatible API.
  */
-final class S3SeekableChannelProvider implements SeekableChannelsProvider {
+@VisibleForTesting
+public final class S3SeekableChannelProvider implements SeekableChannelsProvider {
 
     /**
      * We always allocate buffers of maximum allowed size for re-usability across reads with different fragment sizes.
      * There can be a performance penalty though if the fragment size is much smaller than the maximum size.
      */
     private static final BufferPool BUFFER_POOL = new BufferPool(S3Instructions.MAX_FRAGMENT_SIZE);
 
+    private static final int DEFAULT_MAX_KEYS_PER_BATCH = 1000;
+    private static int maxKeysPerBatch = DEFAULT_MAX_KEYS_PER_BATCH;
+
+    @VisibleForTesting
+    public static void setMaxKeysPerBatch(final int maxKeysPerBatch) {
+        S3SeekableChannelProvider.maxKeysPerBatch = maxKeysPerBatch;
+    }
```

Comment:
I'm not sure it's worth publicly exposing these sorts of test-only configuration levers for physical S3 testing.

We could test this with localstack / minio locally by having path w/ more than 1000 items if we wanted.

Reply:
Got it, I will add localstack tests instead.